Microsoft.Toolkit.Parsers/Markdown/MarkdownDocument.cs

```diff
@@ -365,4 +368,4 @@ public override string ToString()
             return string.Join("\r\n", Blocks);
```

Comment:
Might not have gotten to it yet.  Is there a test that checks that the output of the MarkdownDocument toString is the same as what was passed in to Parse?

Reply:
No, that was before my PR :)

Reply:
Might be good to think about adding another test then that has a sample markdown text blob, gets read by the parser, and then the document gets serialized back to markdown and compare the results?

Reply:
There are numerous cases where this isn't done, or the returned ToString isn't the same is the input. Examples are the `CodeBlock` where it just returns the text. `HorizontalRuleBlock` only returns one kind of raw line, when there are several ways to do it, etc.

It could be done, but it would take some considerable reworking, possibly for a future PR like #1767 or future PRs.